examples/benchmarks/bubblebench/simple-tree/src/appState.ts

```diff
@@ -0,0 +1,76 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import {
+	type IAppState,
+	type IBubble,
+	type IClient,
+	makeBubble,
+	randomColor,
+} from "@fluid-example/bubblebench-common";
+import type { InsertableTypedNode, TreeView } from "@fluidframework/tree/internal";
+import { type App, type Bubble, Client } from "./schema.js";
+
+export class AppState implements IAppState {
+	readonly localClient: Client;
+
+	constructor(
+		private readonly tree: TreeView<App>,
+		public width: number,
+		public height: number,
+		numBubbles: number,
+	) {
+		this.tree.root.clients.insertAtEnd(
+			this.createInitialClientNode(numBubbles) as unknown as Client,
+		);
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		const lastClient = this.tree.root.clients.at(this.tree.root.clients.length - 1)!;
+		this.localClient = lastClient;
+
+		console.log(
+			`created client with id ${this.localClient.clientId} and color ${this.localClient.color}`,
+		);
+	}
+
+	public applyEdits() {}
+
+	createInitialClientNode(numBubbles: number): IClient {
+		const bubbles: IBubble[] = [];
+		// create and add initial bubbles to initial client json tree
+		for (let i = 0; i < numBubbles; i++) {
+			const bubble = makeBubble(this.width, this.height);
+			bubbles.push(bubble);
+		}
+
+		const client: IClient = {
+			clientId: `${Math.random()}`,
+			color: randomColor(),
+			bubbles,
+		};
+
+		return client;
+	}
+
+	public get clients() {
+		return [...this.tree.root.clients];
+	}
+
+	public setSize(width?: number, height?: number) {
+		this.width = width ?? 640;
+		this.height = height ?? 480;
+	}
+
+	public increaseBubbles() {
+		this.localClient.bubbles.insertAtEnd(
+			makeBubble(this.width, this.height) as InsertableTypedNode<typeof Bubble>,
```

Comment:
This cast shouldn't be needed. If it is, we probably have a bug we need to fix.
```suggestion
			makeBubble(this.width, this.height),
```